libp2p/utils.py

```diff
@@ -73,9 +73,8 @@ def encode_delim(msg: bytes) -> bytes:
 
 async def read_delim(reader: Reader) -> bytes:
     msg_bytes = await read_varint_prefixed_bytes(reader)
-    # TODO: Investigate if it is possible to have empty `msg_bytes`
-    if len(msg_bytes) != 0 and msg_bytes[-1:] != b"\n":
-        raise ValueError(f'msg_bytes is not delimited by b"\\n": msg_bytes={msg_bytes}')
+    if len(msg_bytes) == 0 or msg_bytes[-1:] != b"\n":
+        raise ParseError(f'msg_bytes is not delimited by b"\\n": msg_bytes={msg_bytes}')
```

Comment:
the message being `0` bytes and the message not ending with a `\n` are _technically_  two different cases it could be handy to discriminate b/t